clang/test/Preprocessor/sycl-offload-macro.cpp

```diff
@@ -0,0 +1,35 @@
+/// Check predefined macros are enabled for SYCL offload device compilation :
+//  RUN:   %clang -fsycl -mavx -x c -dM -E %s -o - | FileCheck -match-full-lines --check-prefix=CHECK-TARG %s
+
+/// Check predefined macros are not enabled for SYCL offload device compilation :
+//  RUN:   %clang -fsycl -x c -dM -E %s -o - | FileCheck %s
+
+/// Check predefined macros are enabled for SYCL offload host compilation :
+//  RUN:   %clang -mavx -x c -dM -E %s -o - | FileCheck -match-full-lines --check-prefix=CHECK-HOST %s
```

Comment:
You can check both sets of macro-definitions for host and device compiles using the first run using `#ifdef __SYCL_DEVICE_ONLY__`.
See original patch https://github.com/llvm/llvm-project/commit/ccac6b2bf877337a883c3763e41a529d8f9cc1ff#diff-8ab6bb1bf31f415b2158d602c2fcb7a4R5 as a reference.

Moveover, I don't think we should invoke the driver here. Driver's job is checked by clang/test/Driver/sycl-offload.cpp test. Instead you can check what macro-definitions are set if `-target-cpu` and `-target-feature` are provided to the front-end.

Reply:
Thanks Alexey for taking a look at the patch and the feedback.  Since we have already checked driver's support here by clang/test/Driver/sycl-offload.cpp test, this test is not needed and i have removed it.